packages/components/src/interfaces/native/ListItem.ts

```diff
@@ -1,6 +1,108 @@
-import { ParagraphProps } from "./Paragraph"
+import { Type, type Static } from "@sinclair/typebox"
+import { ParagraphSchema } from "./Paragraph"
+import { BaseOrderedListSchema } from "./OrderedList"
+import { BaseUnorderedListSchema } from "./UnorderedList"
 
-export interface ListItemProps {
-  type: "listItem"
-  content: ParagraphProps[]
-}
+export const BaseListItemSchema = Type.Object({
+  type: Type.Literal("listItem"),
+})
+
+// Note: The content key is a recursive schema and the definitions from both
+// OrderedList and UnorderedList are duplicated here to prevent circular
+// dependencies
+// FIXME: This gives a circular dependency error
+export const ListItemSchema = Type.Recursive(
+  (listItemSchema) =>
+    Type.Composite([
+      BaseListItemSchema,
+      Type.Object({
+        content: Type.Array(
+          Type.Union([
+            ParagraphSchema,
+            Type.Composite([
+              BaseOrderedListSchema,
+              Type.Object({
+                content: Type.Array(listItemSchema),
+              }),
+            ]),
+            Type.Composite([
+              BaseUnorderedListSchema,
+              Type.Object({
+                content: Type.Array(listItemSchema),
+              }),
```

Comment:
Hmm the context in which the `content` key is used here is different between the ordered and unordered lists, so I think it might not be possible to bump it one level up? Especially since Paragraph don't use the same type of `content` here.